crates/biome_js_analyze/src/lint/nursery/no_useless_undefined_initialization.rs

```diff
@@ -0,0 +1,139 @@
+use biome_analyze::{
+    context::RuleContext, declare_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource,
+};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_syntax::JsVariableDeclaration;
+use biome_rowan::{AstNode, BatchMutationExt, TextRange};
+
+use crate::JsRuleAction;
+
+declare_rule! {
+    /// Disallow initializing variables to undefined.
+    ///
+    /// A variable that is declared and not initialized to any value automatically gets the value of undefined.
+    /// It’s considered a best practice to avoid initializing variables to undefined.
+    ///
+    /// Source: https://eslint.org/docs/latest/rules/no-undef-init
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// var a = undefined;
+    ///
+    /// let b = undefined, c = 1, d = undefined;
+    ///
+    /// for (let i = 0; i < 100; i++) {
+    /// 	let i = undefined;
+    /// }
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```js
+    /// var a = 1;
+    /// class Foo {
+    /// 	bar = undefined;
+    /// }
+    /// ```
+    ///
+    pub NoUselessUndefinedInitialization {
+        version: "next",
+        name: "noUselessUndefinedInitialization",
+        sources: &[RuleSource::Eslint("no-undef-init")],
+        fix_kind: FixKind::Safe,
+        recommended: false,
+    }
+}
+
+impl Rule for NoUselessUndefinedInitialization {
+    type Query = Ast<JsVariableDeclaration>;
+    type State = (String, TextRange);
+    type Signals = Vec<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+        let let_or_var_kind = node.is_let() || node.is_var();
+
+        let mut signals = vec![];
+
+        if !let_or_var_kind {
+            return signals;
+        }
+
+        for declarator in node.declarators() {
+            let decl = match declarator {
+                Ok(item) => item,
+                _ => continue,
+            };
+
+            let initializer = match decl.initializer() {
+                Some(init) => init,
+                _ => continue,
+            };
+
+            let expression = match initializer.expression() {
+                Ok(expr) => expr,
+                _ => continue,
+            };
```

Comment:
All of this can be chained using a combination of `ok()`,  `.and_then()` and `.map()`.
It that's not possible, you can use the [`let-else`](https://doc.rust-lang.org/beta/rust-by-example/flow_control/let_else.html) syntax:

```rs
let Ok(decl) = declarator else {
	continue
}
```